provider/pkg/googleclient/retry_transport.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2016-2022, Pulumi Corporation.
+
+// Adapted from https://github.com/GoogleCloudPlatform/magic-modules/blob/e9a0e8f3ca4efe711c2db0b12846623822120de0/mmv1/third_party/terraform/utils/retry_transport.go
+
+package googleclient
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	"github.com/hashicorp/errwrap"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
```

Comment:
It looks like this is only used to define `RetryError` but is pulling in a bunch of extra deps as a result. I'd prefer if we could trim that down since most of this implementation is downstream now anyway.

Reply:
Done. I validated that the example [here](https://github.com/pulumi/pulumi-google-native/issues/256#issuecomment-995027982) now passes.